tools/src/main/java/org/apache/kafka/tools/consumer/ConsoleConsumerOptions.java

```diff
@@ -0,0 +1,414 @@
+     *   2. Explicit --from-beginning given -> 'earliest'
+     *   3. Default value of 'latest'
+     * In case both --from-beginning and an explicit value are specified an error is thrown if these
+     * are conflicting.
+     */
+    private void setAutoOffsetResetValue(Properties props) {
+        String earliestConfigValue = "earliest";
+        String latestConfigValue = "latest";
+
+        if (props.containsKey(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG)) {
+            // auto.offset.reset parameter was specified on the command line
+            String autoResetOption = props.getProperty(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG);
+            if (fromBeginning() && !earliestConfigValue.equals(autoResetOption)) {
+                // conflicting options - latest und earliest, throw an error
+                System.err.println("Can't simultaneously specify --from-beginning and 'auto.offset.reset=" + autoResetOption + "', " +
+                        "please remove one option");
+                Exit.exit(1);
+            }
+            // nothing to do, checking for valid parameter values happens later and the specified
+            // value was already copied during .putall operation
+        } else {
+            // no explicit value for auto.offset.reset was specified
+            // if --from-beginning was specified use earliest, otherwise default to latest
+            String autoResetOption = fromBeginning() ? earliestConfigValue : latestConfigValue;
+            props.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, autoResetOption);
+        }
+    }
+
+    private long parseOffset() {
+        if (options.has(offsetOpt)) {
+            switch (options.valueOf(offsetOpt).toLowerCase(Locale.ROOT)) {
+                case "earliest":
+                    return ListOffsetsRequest.EARLIEST_TIMESTAMP;
+                case "latest":
+                    return ListOffsetsRequest.LATEST_TIMESTAMP;
+                default:
+                    String offsetString = options.valueOf(offsetOpt);
+                    try {
+                        long offset = Long.parseLong(offsetString);
+                        if (offset < 0) {
+                            invalidOffset(offsetString);
+                        }
+                        return offset;
+                    } catch (NumberFormatException nfe) {
+                        invalidOffset(offsetString);
+                    }
+            }
+        } else if (fromBeginning()) {
+            return ListOffsetsRequest.EARLIEST_TIMESTAMP;
+        }
+        return ListOffsetsRequest.LATEST_TIMESTAMP;
+    }
+
+    private void invalidOffset(String offset) {
+        CommandLineUtils.printUsageAndExit(parser, "The provided offset value '" + offset + "' is incorrect. Valid values are " +
+                "'earliest', 'latest', or a non-negative long.");
+    }
+
+    private MessageFormatter buildFormatter() {
+        MessageFormatter formatter = null;
```

Comment:
Yes it's needed otherwise the compiler complains `formatter` may not have been initialized as it does not understand the catch clause causes the tool to exit.